### PR TITLE
Conversions: Pluralize aliases + add generic speed triggers

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

Pluralizes the speed aliases in the UI. Adds generic speed triggers.

## People to notify

@bsstoner 
cc/ @mintsoft 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/conversions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Conversions.pm [hunk 1]
```diff
@@ -61,7 +61,7 @@ my $maximum_input = 10**100;
 handle query => sub {
 
     # for natural language queries, settle with default template / data
-    if (exists($lang_triggers{$_}) && $_=~ m/(angle|area|(?:digital storage)|duration|energy|force|mass|power|pressure|temperature|volume)/) {
+    if (exists($lang_triggers{$_}) && $_=~ m/(angle|area|(?:digital storage)|duration|energy|force|mass|power|pressure|speed|temperature|volume)/) {
```
Comment (mintsoft) [inline lib/DDG/Goodie/Conversions.pm]: Can we support "Velocity" as well?

Comment (pjhampton) [inline lib/DDG/Goodie/Conversions.pm]: Yeah, this is a pretty naive solution. I'm going to scale this up. Please don't merge for now 👍

**Comment by pjhampton** [comment]
Is the cpan dep `List::Flatten` already installed in the backend? I can't see it in use in any of the zci repos cc @jbarrett

File: lib/DDG/Goodie/Conversions.pm [hunk 2]
```diff
@@ -27,12 +32,38 @@ foreach my $type (@types) {
...
+        push(@natural_language_triggers, "$key $_");
+    }
+}
+
+##
+## Declares the triggering scheme
+##
+
+triggers any => @triggers;
+triggers any => @natural_language_triggers;
```
Comment (moollaza) [inline lib/DDG/Goodie/Conversions.pm]: `triggers any => (@triggers, @natural_language_triggers)`

I believe that works too 👍

File: lib/DDG/Goodie/Conversions.pm [hunk 3]
```diff
@@ -27,12 +32,38 @@ foreach my $type (@types) {
...
+my @generics = qw/calculator converter conversion conversions/;
 
-triggers any => @lang_triggers;
-my %lang_triggers = map { $_ => 1 } @lang_triggers;
+# appends the above generics to the /values/ in the yml file
+# unit -> unit converter, unit calculator, ...
+for my $array (values %natlang_hash) {
+    for my $val (@$array) {
+        foreach(@generics) {
+            push(@natural_language_triggers, "$val $_");
```
Comment (moollaza) [inline lib/DDG/Goodie/Conversions.pm]: `push @natural_language_triggers, map { "$val $_" } @generics;`

Slightly more Perl-y approach 👍

File: lib/DDG/Goodie/Conversions.pm [hunk 4]
```diff
@@ -27,12 +32,38 @@ foreach my $type (@types) {
...
+        }
+    }
+}
+
+# appends the above generics to the /keys/ in the yml file
+# length -> length converter, length calculator, ...
+for my $key (keys %natlang_hash) {
+    foreach(@generics) {
+        push(@natural_language_triggers, "$key $_");
```
Comment (moollaza) [inline lib/DDG/Goodie/Conversions.pm]: same here,  `push @natural_language_triggers, map {"$val $_"} @generics;`

File: lib/DDG/Goodie/Conversions.pm [hunk 5]
```diff
@@ -58,13 +89,44 @@ sub magnitude_order {
...
+}
+
+# checks the base of the query
+# eg. velocity converter --> speed
+sub get_base_information {
+    my $input = shift;
+    $input =~ s/calculator|converter|conversion\s?|\s//gi;
+
+    foreach my $key (keys %natlang_hash) {
+        if($key eq $input) {
```
Comment (moollaza) [inline lib/DDG/Goodie/Conversions.pm]: `return $key if $key eq input`

Comment (mintsoft) [inline lib/DDG/Goodie/Conversions.pm]: I think something like `return if grep {$_ eq $input}, keys %natlang_hash` should work also

File: lib/DDG/Goodie/Conversions.pm [hunk 6]
```diff
@@ -58,13 +89,44 @@ sub magnitude_order {
...
+            return $key;
+        }
+    }
+
+    foreach my $key (keys %natlang_hash) {
+        if(exists $natlang_hash{$key}) {
```
Comment (mintsoft) [inline lib/DDG/Goodie/Conversions.pm]: I'd invert this to 'next unless exists $natlang_hash{$key};`

File: lib/DDG/Goodie/Conversions.pm [hunk 7]
```diff
@@ -58,13 +89,44 @@ sub magnitude_order {
...
+            my @hash_kv = @{$natlang_hash{$key}};
+            foreach my $value (@hash_kv) {
```
Comment (mintsoft) [inline lib/DDG/Goodie/Conversions.pm]: Again I think `grep` is a nice solution here

Comment (mintsoft) [inline lib/DDG/Goodie/Conversions.pm]: I think there's a nicer way of doing this entire section, but I can't quite put my finger on it atm

**Comment by pjhampton** [comment]
Nice review @moollaza @mintsoft 💯 

@mintsoft we were discussing yesterday dedicating a half-week/week to refactoring the 3 conversion IAs (this, crypto, currency) once we get them stable and optimize the triggering in the coming weeks 👍  I got a bit carried away with this one, but we will revisit soon.

**Comment by moollaza** [comment]
@pjhampton Thanks! This LGTM :+1:

**Comment by bsstoner** [comment]
This was released to production 🚀

File: share/goodie/conversions/langTriggers.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+length:
+  - online
+  - unit
+  - unit
```
Comment (moollaza) [inline share/goodie/conversions/langTriggers.yml]: whoops, looks like a duplicate

**Pull request merged**